src/components/Layout/Header/Header.tsx

```diff
@@ -98,19 +100,31 @@ export const Header = memo(() => {
 
   const currentWalletId = useSelector(selectWalletId)
   const walletAccountIds = useSelector(selectWalletAccountIds)
-  const hasUtxoAccountIds = useMemo(
-    () => walletAccountIds.some(accountId => isUtxoAccountId(accountId)),
-    [walletAccountIds],
-  )
 
   useEffect(() => {
     const isMetaMaskMultichainWallet = wallet instanceof MetaMaskShapeShiftMultiChainHDWallet
     if (!(currentWalletId && isMetaMaskMultichainWallet && isSnapInstalled === false)) return
 
-    // We have just detected that the user doesn't have the snap installed currently
-    // We need to check whether or not the user had previous non-EVM AccountIds and clear those
-    if (hasUtxoAccountIds) appDispatch(portfolio.actions.clearWalletMetadata(currentWalletId))
-  }, [appDispatch, currentWalletId, hasUtxoAccountIds, isSnapInstalled, wallet, walletAccountIds])
+    const nonEvmAccountIds = walletAccountIds.filter(
+      accountId => !isEvmChainId(fromAccountId(accountId).chainId),
+    )
+
+    // We have just detected that the user doesn't have the snap installed currently, so disable all non-evm accounts
+    nonEvmAccountIds.forEach(accountId =>
+      appDispatch(portfolio.actions.disableAccountId(accountId)),
+    )
+
+    // close the account management model if it's open (since account management isn't enabled for non-snaps metamask)
+    if (isManageAccountsOpen) closeManageAccountsModal()
```

Comment:
On the other hand in the edge cases side of things, while the previously mentioned edge case (if I'm right and the bug is present) is more of a symptom of something we probably don't want, the explicit handling of this edge case is something I'm wondering whether or not we want:

This feels like a case of us trying to handle every possible edge case in the app, at the expense of code quality (i.e a bunch of hacked around conditions for every single possible scenario).
Same comment here as with the previous lending default asset things on chain support loss (see this comment https://github.com/shapeshift/web/pull/7029#discussion_r1622042222 - as previously mentioned, comes as additional code, whilst only handling *some* of the cases), this is a scenario a user wouldn't get into, except if the user is ops trying to break the app:

- for lending, why would a user disconnect the account for the asset they're interacting with?
- here for MM with multichain snap, why would a user disconnect the snap - and nuke their multi-account/non-EVM chain capabilities, while in the process of account management?

Not a this PR issue per se and more of a conceptual concern from a product <-> eng. perspective, but personally in favour of the path of least code and hacked around conditions to keep things clean and avoid potential bugs.

tl;dr more safe sometimes is less safe

Reply:
@NeOMakinG 💯

> Users can sometimes act unpredictably or make mistakes. Rather than closing modals or disabling the UI entirely

Indeed, which all comes down to the definition of what constitutes a bug IMO. i.e for borrow/repayment asset disconnected, is keeping their selection a bug? You'd assume not, they've already selected that asset as they want to interact with it. Assuming they yolo their account/s for that asset out (weird, but indeed, unpredictable users) then they can't interact with it anymore, but doesn't mean we should void their selection. The UI handles this gracefully (ish, they can't get a new quote for that specific asset but the UI still works as it should), and if they want to switch to an asset they can interact with (which is what they're probably going to do if they disconnected their account for the previous asset), they can, we don't need to automagically set a default on them.

> I will never be in favor of closing modals without telling anything but more in favor of changing its content to inform the user that he can't use this feature because of its current account state, because he initially chosen to open this modal

And precisely, as far as disabling the feature (account management modal) and informing the user that loading more accounts isn't possible because of lack of BIP44 support, we already do this holistically... meaning, including for the multichain snap! 
In other words, we already have the holistic solution, and don't need the granular, hacky, and automagical (meaning without the user understanding what's happening and why) edge-case handling